### PR TITLE
Feat/text highlight refactor

### PR DESCRIPTION
**Pull request by M-i-k-e-l**
## Description
Text - highlight - refactor and handle some edge cases (add tests and move to utils)

## Changelog
Text - highlight - refactor and handle some edge cases

### CODE REVIEW

File: src/utils/__tests__/textUtils.spec.js [hunk 1]
```diff
@@ -45,10 +45,15 @@ describe('Text', () => {
...
-    it('Should handle empty string .', () => {
+    it('Should handle empty string.', () => {
       const result = getTextPartsByHighlight('@ancing in the @ark', '');
       expect(result).toEqual([{string: '@ancing in the @ark', shouldHighlight: false}]);
     });
+
+    it('Should handle full string.', () => {
+      const result = getTextPartsByHighlight('Dancing in the Dark', 'Dancing in the Dark');
+      expect(result).toEqual([{string: 'Dancing in the Dark', shouldHighlight: true}]);
+    });
```
Comment (lidord-wix) [inline src/utils/__tests__/textUtils.spec.js:83]: Consider adding a test for more than full string: 
`getTextPartsByHighlight('Dancing in the Dark', 'Dancing in the Darkk');`

**Review by lidord-wix — COMMENTED** [review]
Looks good and makes a lot of sense!
I left small comments :)

File: src/components/text/__tests__/index.spec.js [hunk 2]
```diff
@@ -55,4 +60,58 @@ describe('Text', () => {
...
+      const result = uut.getArrayPartsByHighlight('!ancing in the !ark', ['!a']);
+      expect(result).toEqual([{string: '!a', shouldHighlight: true}, {string: 'ncing in the !ark', shouldHighlight: false}]);
+    });
+
+    it('Should handle special characters starts with @', () => {
+      const uut = new Text({});
+      const result = uut.getArrayPartsByHighlight('uilib@wix.com', ['@wix']);
+      expect(result).toEqual([{string: 'uilib', shouldHighlight: false}, {string: '@wix', shouldHighlight: true}, {string: '.com', shouldHighlight: false}]);
+    });
+  });
```
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply src/components/text/__tests__/index.spec.js]: I don't know, I did not write it; I'm guessing that there were bugs that needed verification.

File: src/utils/__tests__/textUtils.spec.js (hunk 1 above)
Comment (M-i-k-e-l) [reply src/utils/__tests__/textUtils.spec.js:83]: I don't think this is very important in the current logic, adding in case it ever changes 👍 
Also noticed we don't have these tests for the array --> found another bug.

**Review by lidord-wix — APPROVED** [review]
